src/BTCPayServer.Lightning.Eclair/EclairLightningClient.cs

```diff
@@ -282,46 +297,61 @@ public async Task<PayResponse> Pay(string bolt11, PayInvoiceParams payParams, Ca
             {
                 return new PayResponse(PayResult.Error, exception.Message);
             }
-
-            return new PayResponse(PayResult.CouldNotFindRoute);
+            catch (Exception exception)
+            {
+                return cts.Token.IsCancellationRequested
+                    ? new PayResponse(PayResult.Unknown)
+                    : new PayResponse(PayResult.Error, exception.Message);
+            }
+            return new PayResponse(PayResult.Unknown);
         }
 
         public async Task<PayResponse> Pay(PayInvoiceParams payParams, CancellationToken cancellation = default)
         {
+            // Pay the invoice - cancel after timeout, potentially caused by hold invoices
+            using var cts = CancellationTokenSource.CreateLinkedTokenSource(cancellation);
+            var timeout = payParams?.SendTimeout ?? PayInvoiceParams.DefaultSendTimeout;
+            cts.CancelAfter(timeout);
+
             try
             {
-                var paymentHash = payParams.PaymentHash.ToString();
                 var req = new SendToNodeRequest
                 {
-                    NodeId = payParams.Destination.ToString(),
+                    NodeId = payParams.Destination?.ToString(),
                     AmountMsat = payParams.Amount?.MilliSatoshi,
-                    PaymentHash = paymentHash,
                     MaxFeePct = payParams.MaxFeePercent != null ? (int)Math.Round(payParams.MaxFeePercent.Value) : null,
                     MaxFeeFlatSat = payParams.MaxFeeFlat?.Satoshi,
 
                 };
-                var uuid = await _eclairClient.SendToNode(req, cancellation);
-                while (!cancellation.IsCancellationRequested)
+                var uuid = await _eclairClient.SendToNode(req, cts.Token);
+                while (!cts.Token.IsCancellationRequested)
                 {
-                    var status = await _eclairClient.GetSentInfo(paymentHash, uuid, cancellation);
+                    var status = await _eclairClient.GetSentInfo(null, uuid, cts.Token);
```

Comment:
We need to use the UUID instead of the payment hash here. Eclair doesn't support defining the payment hash for keysend payments.